core/src/cluster_info.rs

```diff
@@ -533,6 +540,22 @@ struct ResponseScore {
     score: u64,             // Relative score of the response
 }
 
+// Retains only CRDS values associated with nodes with enough stake.
+// (ContactInfo are exempt)
+fn retain_staked(values: &mut Vec<CrdsValue>, stakes: &HashMap<Pubkey, u64>) {
+    values.retain(|value| {
+        match value.data {
+            // TODO: Confirm that NodeInstance does not need to be exempted.
+            // TODO: What about Version?
```

Comment:
Hmm, I think Vote/EpochSlots/SnapshotHashes should be retained as well because:
* Vote - I'm worried it'll impact new validators starting up without any stake yet
*  EpochSlots - unstaked nodes can still help repair
*  SnapshotHashes - unstaked nodes can still serve snapshots

Version would be nice to have as well, otherwise unstaked voting nodes will show up with no version in the various dashboards.   At this point I think we've practically included all the big ones back into gossip, so perhaps this approach just isn't going to work?

Reply:
Added those to the exceptions. This is still excluding `DuplicateShreds` and few others which are 500+ values per pubkey. So might still have some use.